corehq/apps/api/es.py

```diff
@@ -421,19 +426,22 @@ def consume_params(self, raw_params):
             )
 
 
-query_param_consumers = [
+xform_param_consumers = [
```

Comment:
For this commit, I started `xform_param_consumers` and `case_param_consumers` as duplicates of the original `query_param_consumers`, then pared each down individually.  To do this, I checked each term against the corresponding mapping to see whether it is a real, searchable, key.  As you might imagine, `xmlns` doesn't apply to case API requests, and `case_name` doesn't apply to xform API requests.

This _is_ a slight back-end behavior change, but one I'm comfortable with.  Previously, someone (inexplicably) filtering the case API by `xmlns` would see a ES query filtered by `xmlns.exact`.  Now, they'll see an ES query filtered just by `xmlns`, and their search term will be coerced to lowercase.  Since neither exists in the case index, both queries will return no results, and the distinction is moot.